logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/VerifyExistingClientUseCase.kt

```diff
@@ -0,0 +1,47 @@
+package com.wire.kalium.logic.feature.client
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.client.Client
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.conversation.ClientId
+import com.wire.kalium.logic.functional.fold
+
+/**
+ * Checks if the given client is still exists on the backend, otherwise returns failure.
+ */
+interface VerifyExistingClientUseCase {
+
+    /**
+     * @param clientId client id of client
+     * @return success if the given id is registered, failure otherwise
+     */
+    suspend operator fun invoke(clientId: ClientId): VerifyExistingClientResult
+}
+
+internal class VerifyExistingClientUseCaseImpl(
+    private val clientRepository: ClientRepository
+) : VerifyExistingClientUseCase {
+
+    override suspend fun invoke(clientId: ClientId): VerifyExistingClientResult {
+        return clientRepository.selfListOfClients()
+            .fold({
+                VerifyExistingClientResult.Failure.Generic(it)
+            }, { listOfClients ->
+                val client = listOfClients.firstOrNull { it.id == clientId }
```

Comment:
This looks like a bug to me. We check that our local client is equal to first client in the list of self clients. Shouldn't this be a `listOfClients.contains({ it.id == clientId })`?

Reply:
yes this must be replaced with `listOfClients.contains({ it.id == clientId })`

Reply:
Tried to write a failing test and it was passing :) 

`firstOrNull` returns first matching element or null. So essentially same behaviour as `istOfClients.contains({ it.id == clientId })`